app/src/features/settings/components/GlobalSettings/components/ImplicitRuleTesting/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { useSelector } from "react-redux";
+import { getAppMode } from "store/selectors";
+import SettingsItem from "../SettingsItem";
+import { StorageService } from "init";
+import ErrorCard from "components/misc/ErrorCard";
+import { RuleTypesOptions } from "./components/RuleTypesOptions";
+import { CONSTANTS as GLOBAL_CONSTANTS } from "@requestly/requestly-core";
+import { isFeatureCompatible } from "utils/CompatibilityUtils";
+import APP_CONSTANTS from "config/constants";
+
+export const ImplicitRuleTesting = () => {
+  const appMode = useSelector(getAppMode);
+  const [isImplicitRuleTestingEnabled, setIsImplicitRuleTestingEnabled] = useState(false);
+  const [enabledRuleTypes, setEnabledRuleTypes] = useState(null);
+  const [widgetVisibility, setWidgetVisibility] = useState(
+    GLOBAL_CONSTANTS.IMPLICIT_RULE_TESTING_WIDGET_VISIBILITY.OFF
+  );
+
+  const isCompatible = useMemo(() => isFeatureCompatible(APP_CONSTANTS.FEATURES.TEST_THIS_RULE), []);
+
+  const handleImplicitRuleTestingToggleChange = useCallback(
+    (status: boolean) => {
+      setIsImplicitRuleTestingEnabled(status);
+
+      const newVisibility = status
+        ? GLOBAL_CONSTANTS.IMPLICIT_RULE_TESTING_WIDGET_VISIBILITY.SPECIFIC
+        : GLOBAL_CONSTANTS.IMPLICIT_RULE_TESTING_WIDGET_VISIBILITY.OFF;
+
+      setWidgetVisibility(newVisibility);
+      StorageService(appMode).saveRecord({
+        [GLOBAL_CONSTANTS.IMPLICIT_RULE_TESTING_WIDGET_CONFIG]: {
+          ruleTypes: enabledRuleTypes,
+          visibility: newVisibility,
+        },
+      });
+    },
+    [appMode, enabledRuleTypes]
+  );
+
+  useEffect(() => {
+    StorageService(appMode)
+      .getRecord(GLOBAL_CONSTANTS.IMPLICIT_RULE_TESTING_WIDGET_CONFIG)
+      .then((data) => {
+        setEnabledRuleTypes(data.ruleTypes);
+        setWidgetVisibility(data.visibility);
+        if (data.visibility === GLOBAL_CONSTANTS.IMPLICIT_RULE_TESTING_WIDGET_VISIBILITY.OFF) {
+          setIsImplicitRuleTestingEnabled(false);
+        } else setIsImplicitRuleTestingEnabled(true);
+      });
```

Comment:
Can we create a helper function for this action?

Reply:
Config is being read only once, how will a util function help here?

Reply:
It is also being read in `useInitImplicitWidgetConfig` hook. Ideally, we should avoid using `StorageService` directly.

Reply:
used the read util function here to read widget config, will update the others after merge